src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.Date.cs

```diff
@@ -0,0 +1,206 @@
+        public static void TestingStringsWithUTCOffsetToDateTimeOffset(string jsonString, string expectedString)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(dataUtf8, isFinalBlock: true, state: default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.String)
+                {
+                    DateTimeOffset expected = DateTimeOffset.ParseExact(expectedString, "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind);
+
+                    Assert.True(json.TryGetDateTimeOffset(out DateTimeOffset actual));
+                    Assert.Equal(expected, actual);
+
+                    Assert.Equal(expected, json.GetDateTimeOffset());
+                }
+            }
+
+            Assert.Equal(dataUtf8.Length, json.BytesConsumed);
+        }
+
+        [Theory]
+        [MemberData(nameof(JsonDateTimeTestData.InvalidISO8601Tests), MemberType = typeof(JsonDateTimeTestData))]
+        public static void TestingStringsInvalidConversionToDateTime(string jsonString)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(dataUtf8, isFinalBlock: true, state: default);
+            while (json.Read())
+            {
+                Assert.False(json.TryGetDateTime(out DateTime actualDateTime));
+                Assert.Equal(default, actualDateTime);
+
+                try
+                {
+                    DateTime value = json.GetDateTime();
+                    Assert.True(false, "Expected GetDateTime to throw FormatException due to invalid ISO 8601 input.");
+                }
+                catch (FormatException)
+                { }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(JsonDateTimeTestData.InvalidISO8601Tests), MemberType = typeof(JsonDateTimeTestData))]
+        public static void TestingStringsInvalidConversionToDateTimeOffset(string jsonString)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(dataUtf8, isFinalBlock: true, state: default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.String)
+                {
+                    Assert.False(json.TryGetDateTimeOffset(out DateTimeOffset actualDateTime));
+                    Assert.Equal(default, actualDateTime);
+
+                    try
+                    {
+                        DateTimeOffset value = json.GetDateTimeOffset();
+                        Assert.True(false, "Expected GetDateTimeOffset to throw FormatException due to invalid ISO 8601 input.");
+                    }
+                    catch (FormatException)
+                    { }
+                }
+            }
+        }
+
+        [Fact]
+        public static void Regression39067_TestingDateTimeMinValue()
```

Comment:
Is it possible to run this in RemoteExecutor and locally modify the timezone used?  Not sure if that's possible or not, I'd imagine we might have similar requirements for DateTime types though.

Reply:
I've tried to do this in the past but was advised that this can affect other tests, or the test might crash and we'll not be able to revert the system settings.

Reply:
> Is it possible to run this in RemoteExecutor and locally modify the timezone used?

Time zone setting is user global setting and not process setting. I wouldn't recommend that.